### PR TITLE
Drop virtctl related tests and add DryRun tests in hotplug.go 

### PR DESCRIPTION
**Pull request by 0xFelix**
### What this PR does

Drop the virtctl related tests in tests/storage/hotplug.go. As long as
virtctl calls the correct API endpoints these tests don't add any value. (see https://github.com/kubevirt/kubevirt/pull/12634)

Verify that the DryRun option in AddVolume requests works as intended.
These cases were covered only indirectly by tests using virtctl previously.

The goal of this and further PRs is to reduce the amount of unnecessary uses of virtctl in the e2e tests to stabilize tests, to reduce the number of flakes and to remove redudant tests.

Fixes #

### Why we need it and why it was done in this way
The following tradeoffs were made:

The following alternatives were considered:

Links to places where the discussion took place: 

### Checklist

This checklist is not enforcing, but it's a reminder of items that could be relevant to every PR.
Approvers are expected to review this list.

- [ ] Design: A [design document](https://github.com/kubevirt/community/tree/main/design-proposals) was considered and is present (link) or not required
- [ ] PR: The PR description is expressive enough and will help future contributors
- [ ] Code: [Write code that humans can understand](https://en.wikiquote.org/wiki/Martin_Fowler#code-for-humans) and [Keep it simple](https://en.wikipedia.org/wiki/KISS_principle)
- [ ] Refactor: You have [left the code cleaner than you found it (Boy Scout Rule)](https://learning.oreilly.com/library/view/97-things-every/9780596809515/ch08.html)
- [ ] Upgrade: Impact of this change on upgrade flows was considered and addressed if required
- [ ] Testing: New code requires [new unit tests](https://github.com/kubevirt/kubevirt/blob/main/docs/reviewer-guide.md#when-is-a-pr-good-enough). New features and bug fixes require at least on e2e test
- [ ] Documentation: A [user-guide update](https://github.com/kubevirt/user-guide/) was considered and is present (link) or not required. You want a user-guide update if it's a user facing feature / API change.
- [ ] Community: Announcement to [kubevirt-dev](https://groups.google.com/g/kubevirt-dev/) was considered

### Release note

```release-note
None
```

### CODE REVIEW

File: tests/storage/hotplug.go [hunk 1]
```diff
@@ -514,6 +515,19 @@ var _ = SIGDescribe("Hotplug", func() {
...
+		vmi, err := virtClient.VirtualMachineInstance(vm.Namespace).Get(context.Background(), vm.Name, metav1.GetOptions{})
+		Expect(err).ToNot(HaveOccurred())
+		libwait.WaitForSuccessfulVMIStart(vmi,
+			libwait.WithTimeout(240),
+		)
+
+		addVolumeFunc(vm.Name, vm.Namespace, "testvolume", dv.Name, v1.DiskBusSCSI, true, "")
+		verifyNoVolumeAttached(vmi, "testvolume")
+	}
+
```
Comment (fossedihelm) [inline tests/storage/hotplug.go:530]: Do we really need this function? Can't we just do it inside the DescribeTable body function?
Also, the function does more than the name expose.
I'd suggest you move the content of this to the DescribeTable. WDYT?

**Review by fossedihelm — COMMENTED** [review]
Thank you @0xFelix!
I'd love to see less use of inner function here :)

File: tests/storage/hotplug.go (hunk 1 above)
Comment (0xFelix) [reply tests/storage/hotplug.go:530]: This night I woke up from a nightmare and then I knew I had to do this. +1

**Comment by fossedihelm** [comment]
/lgtm
Thank you!

**Comment by 0xFelix** [comment]
/test pull-kubevirt-check-tests-for-flakes

Failed because of a setup error.

**Pull request merged**